src/ValidationError.php

```diff
@@ -2,23 +2,32 @@
 
 namespace Schemer;
 
-class ValidationError
+final class ValidationError
 {
+    const NOT_ALPHANUMERIC = 'NOT_ALPHANUMERIC';
+    const NOT_EMAIL = 'NOT_EMAIL';
+    const LENGTH_MISMATCH = 'LENGTH_MISMATCH';
+    const NOT_LOWERCASE = 'NOT_LOWERCASE';
+    const NOT_UPPERCASE = 'NOT_UPPERCASE';
+    const TOO_LONG = 'TOO_LONG';
+    const TOO_SHORT = 'TOO_SHORT';
+    const REGEX_MISMATCH = 'REGEX_MISMATCH';
```

Comment:
Yeah, not sure it's necessary but I think I'd prefer to use a constant in code consuming the library than use the literal string.

It does mean this class is going to get a ridiculous number of these though.